hive/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -175,7 +190,12 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
           EnvironmentContext envContext = new EnvironmentContext(
               ImmutableMap.of(StatsSetupConst.DO_NOT_UPDATE_STATS, StatsSetupConst.TRUE)
           );
-          client.alter_table(database, tableName, tbl, envContext);
+
+          try {
+            alterTableMethod.invoke(client, database, tableName, tbl, envContext);
+          } catch (Exception e) {
+            throw new RuntimeMetaException(e, "Failed to alter table for %s.%s", database, tableName);
+          }
```

Comment:
If you use `DynMethods` then you'd call the method like this:

```java
    try {
      clients.run(client -> ALTER_TABLE.invoke(client, null, null, null, null));
    } catch (TException e) {
      throw new RuntimeMetaException(e, "Failed to alter table");
    } catch (InterruptedException e) {
      Thread.currentThread().interrupt();
      throw new RuntimeException("Interrupted in call to alterTable", e);
    }
```

Reply:
With this Iceberg will complain about "CyclomaticComplexity" issue, I think it is probably fine to let the existing exceptional handler to handle the exceptions.